File: StockView-UIUX/src/app/services/user.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { User } from "../models/users";
import { environment } from 'src/environments/environment';
import { PasswordChange } from '../models/passwordChange';

@Injectable({
    providedIn: "root"
})
export class UserService {
    url = environment.host+"user-service/users/";
    constructor(private httpClient: HttpClient) {}

    getAllUsers(): Observable<User[]> {
        return this.httpClient.get<User[]>(this.url);
    }

    getUserById(id: number): Observable<User> {
        return this.httpClient.get<User>(this.url + id);
    }

    emailExists(email: string): Observable<boolean> {
        return this.httpClient.get<boolean>(this.url + "open/emailExists/" + email);
    }

    usernameExists(username: string): Observable<boolean> {
        return this.httpClient.get<boolean>(this.url + "open/usernameExists/" + username);
    }

    isUserActive(username: string): Observable<boolean> {
        return this.httpClient.get<boolean>(this.url + "open/isUserActive/" + username);
    }
 
    getUserByUsernameAndPassword(
        username: string,
        password: string
    ): Observable<User> {
        return this.httpClient.get<User>(this.url + "findByUsernameAndPassword/" + username+"/"+password);
    }
    getUserByUsername(username: string): Observable<User> {
        return this.httpClient.get<User>(this.url + "findByUsername/" + username);
    }

    registerUser(user: User): Observable<User> {
        return this.httpClient.post<User>(this.url+"open/addUser", user);
    }

    deleteUser(id: number): Observable<User> {
        return this.httpClient.delete<User>(this.url + id);
    }

    updateUser(user: User): Observable<User> {
        return this.httpClient.put<User>(this.url, user);
    }

    updatePassword(request: PasswordChange): Observable<boolean> {
        return this.httpClient.put<boolean>(this.url+"updatePassword", request);
    }

    activateUser(email: string): Observable<boolean> {
        return this.httpClient.put<boolean>(this.url + "open/activate", email);
    }
}

File: StockView - Microservices/user-service/src/main/java/com/cts/training/stockview/userservice/service/UserService.java
package com.cts.training.stockview.userservice.service;

import java.util.List;
import java.util.NoSuchElementException;

import com.cts.training.stockview.userservice.model.PasswordChangeRequest;
import com.cts.training.stockview.userservice.model.User;

public interface UserService {

	public List<User> getAllUsers();

	public User getUserById(int id);

	public User getUserByUsernameAndPassword(String username, String password);

	public User addUser(User user);

	public boolean activateUser(String email);

	public User updateUser(User user);

	public void deleteUser(int id);

	User getUserByUsername(String username);

	boolean emailExists(String email) throws NoSuchElementException;

	boolean usernameExists(String email) throws NoSuchElementException;

	public Boolean isUserActive(String username);

	public boolean updatePassword(PasswordChangeRequest request);
}